File: gravitee-am-ui/src/app/services/tag.service.ts
/*
 * Copyright (C) 2015 The Gravitee team (http://gravitee.io)
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *         http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { Injectable } from '@angular/core';
import { AppConfig } from '../../config/app.config';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';

@Injectable()
export class TagService {
  private tagsURL = AppConfig.settings.organizationBaseURL + '/tags/';

  constructor(private http: HttpClient) {}

  list(): Observable<any> {
    return this.http.get<any>(this.tagsURL);
  }

  get(id): Observable<any> {
    return this.http.get<any>(this.tagsURL + id);
  }

  create(tag): Observable<any> {
    return this.http.post<any>(this.tagsURL, tag);
  }

  update(id, tag): Observable<any> {
    return this.http.put<any>(this.tagsURL + id, {
      name: tag.name,
      description: tag.description,
    });
  }

  delete(id): Observable<any> {
    return this.http.delete<any>(this.tagsURL + id);
  }
}

File: gravitee-am-management-api/gravitee-am-management-api-service/src/main/java/io/gravitee/am/management/service/TagService.java
/**
 * Copyright (C) 2015 The Gravitee team (http://gravitee.io)
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *         http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package io.gravitee.am.management.service;

import io.gravitee.am.identityprovider.api.User;
import io.gravitee.am.model.Tag;
import io.gravitee.am.service.model.NewTag;
import io.gravitee.am.service.model.UpdateTag;
import io.reactivex.rxjava3.core.Completable;
import io.reactivex.rxjava3.core.Flowable;
import io.reactivex.rxjava3.core.Maybe;
import io.reactivex.rxjava3.core.Single;

/**
 * @author David BRASSELY (david.brassely at graviteesource.com)
 * @author GraviteeSource Team
 */
public interface TagService {

    Maybe<Tag> findById(String id, String organizationId);

    Flowable<Tag> findAll(String organizationId);

    Single<Tag> create(NewTag tag, String organizationId, User principal);

    Single<Tag> update(String tagId, String organizationId, UpdateTag tag, User principal);

    Completable delete(String tagId, String organizationId, User principal);
}
